### PR TITLE
Gitlab support

### PR DESCRIPTION
**Pull request by jayvdb**
Demo build at https://jayvdb.gitlab.io/community/

One known problem is that second level sub-directory pages do not work.

Pipelines at https://gitlab.com/jayvdb/community/pipelines

### CODE REVIEW

**Comment by jayvdb** [comment]
Hoping this https://github.com/coala/coala-bears/issues/2712 solves the duplicate key problem.

File: community/tests/test_git.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import unittest
+
+from community.git import get_config_remote
+from community.git import get_deploy_url
+from community.git import get_org_name
+from community.git import get_remote_url
```
Comment (sks444) [inline community/tests/test_git.py]: Why not do?
```python
from community.git import (
    get_config_remote,
    get_deploy_url,
    get_org_name,
    ...
)
```

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  script:
+    - ./.ci/build.sh
+
+pages:
+  stage: deploy
+  variables:
+    # Simulate the netlify production flag
+    CONTEXT: production
+  script:
+    - echo 'Nothing to do'
```
Comment (li-boxuan) [inline .gitlab-ci.yml]: Is `script` mandatory?

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+image: python:3.6
+
+stages:
+  - build
+  - deploy
+
```
Comment (li-boxuan) [inline .gitlab-ci.yml]: `test` stage is not added?

File: .gitlab-ci.yml (hunk 2 above)
Comment (jayvdb) [reply .gitlab-ci.yml]: Yup.

> Error: jobs:pages script can't be blank

File: community/tests/test_git.py (hunk 1 above)
Comment (jayvdb) [reply community/tests/test_git.py]: That style consumes two extra lines, and depends on ppl remembering to use "trailing commas", which coala dont have a linter for. (isort may be able to enforce this, but isort has other problems)

**Comment by jayvdb** [comment]
ack https://github.com/coala/community/pull/202/commits/25a82d34e1abc93070f93af74ef0dfa1bf15cd6b https://github.com/coala/community/pull/202/commits/00b097bddadd4f2dfe5a61daf09d1c81b3811168 https://github.com/coala/community/pull/202/commits/ef907f3949a2b8f541d5571635f83748144c8b67 https://github.com/coala/community/pull/202/commits/97cae73cbedfd1934a9d2ee31a648e2b0e8965e4 https://github.com/coala/community/pull/202/commits/d9a344c4b73ded3e075cd3ff8fc54a4b697ea5ac https://github.com/coala/community/pull/202/commits/5933dc8d983c2fbbd27223c59b8a23fa715e5425

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**